test/interval.jl

```diff
@@ -887,4 +887,119 @@
             end
         end
     end
+
+    @testset "floor" begin
+        @test_throws UndefKeywordError floor(Interval(0.0, 1.0))
+
+        @test floor(Interval(0.0, 1.0), on=LeftEndpoint) == Interval(0.0, 1.0)
+        @test floor(Interval(0.5, 1.0), on=LeftEndpoint) == Interval(0.0, 0.5)
+        @test floor(Interval(0.0, 1.5), on=LeftEndpoint) == Interval(0.0, 1.5)
+        @test floor(Interval(0.5, 1.5), on=LeftEndpoint) == Interval(0.0, 1.0)
+
+        @test floor(Interval(0.0, 1.0), on=RightEndpoint) == Interval(0.0, 1.0)
+        @test floor(Interval(0.5, 1.0), on=RightEndpoint) == Interval(0.5, 1.0)
+        @test floor(Interval(0.0, 1.5), on=RightEndpoint) == Interval(-0.5, 1.0)
+        @test floor(Interval(0.5, 1.5), on=RightEndpoint) == Interval(0.0, 1.0)
+
+        # Test supplying a period to floor to
+        interval = Interval(DateTime(2011, 2, 1, 6), DateTime(2011, 2, 2, 18))
+        expected = Interval(DateTime(2011, 2, 1), DateTime(2011, 2, 2, 12))
+        @test_throws UndefKeywordError floor(interval, Day)
+        @test floor(interval, Day; on=LeftEndpoint) == expected
+        @test floor(interval, Day(1); on=LeftEndpoint) == expected
+
+        expected = Interval(DateTime(2011, 1, 31, 12), DateTime(2011, 2, 2))
+        @test floor(interval, Day; on=RightEndpoint) == expected
+        @test floor(interval, Day(1); on=RightEndpoint) == expected
+
+        # Test unbounded intervals
+        @test floor(Interval{Closed, Unbounded}(0.0, nothing), on=LeftEndpoint) == Interval{Closed, Unbounded}(0.0, nothing)
+        @test floor(Interval{Closed, Unbounded}(0.5, nothing), on=LeftEndpoint) == Interval{Closed, Unbounded}(0.0, nothing)
+        @test floor(Interval{Unbounded, Closed}(nothing, 1.0), on=LeftEndpoint) == Interval{Unbounded, Closed}(nothing, 1.0)
+        @test floor(Interval{Unbounded, Closed}(nothing, 1.5), on=LeftEndpoint) == Interval{Unbounded, Closed}(nothing, 1.5)
+        @test floor(Interval{Unbounded, Unbounded}(nothing, nothing), on=LeftEndpoint) == Interval{Unbounded, Unbounded}(nothing, nothing)
+
+        @test floor(Interval{Closed, Unbounded}(0.0, nothing), on=RightEndpoint) == Interval{Closed, Unbounded}(0.0, nothing)
+        @test floor(Interval{Closed, Unbounded}(0.5, nothing), on=RightEndpoint) == Interval{Closed, Unbounded}(0.5, nothing)
+        @test floor(Interval{Unbounded, Closed}(nothing, 1.0), on=RightEndpoint) == Interval{Unbounded, Closed}(nothing, 1.0)
+        @test floor(Interval{Unbounded, Closed}(nothing, 1.5), on=RightEndpoint) == Interval{Unbounded, Closed}(nothing, 1.0)
+        @test floor(Interval{Unbounded, Unbounded}(nothing, nothing), on=RightEndpoint) == Interval{Unbounded, Unbounded}(nothing, nothing)
+    end
+
+    @testset "ceil" begin
+        @test_throws UndefKeywordError ceil(Interval(0.0, 1.0))
+
+        @test ceil(Interval(0.0, 1.0), on=LeftEndpoint) == Interval(0.0, 1.0)
+        @test ceil(Interval(0.5, 1.0), on=LeftEndpoint) == Interval(1.0, 1.5)
```

Comment:
I'm not sure `on` is the right keyword as it makes me think only the left would be changed. I'll think about a better name

Reply:
Thinking about it more I think this is just my personal bias and `on` is probably correct